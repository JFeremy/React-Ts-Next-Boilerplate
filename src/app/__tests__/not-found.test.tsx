import { render, screen } from '@testing-library/react';

import NotFound from '../not-found';

describe('NotFound', () => {
  it('shoud have expected text', async () => {
    render(<NotFound />); // ARRANGE

    const text = screen.getByText('⛔ 404 - Page Not Found ⛔');
    const links = screen.getAllByRole('link');

    expect(text).toBeInTheDocument();
    expect(links).toHaveLength(1);
    expect(links[0]).toHaveProperty('href', 'http://localhost/');
    expect(links[0]).toHaveProperty('text', 'Homepage');
  });

  it('component matches snapshot', async () => {
    const view = render(<NotFound />);
    expect(view.container).toMatchSnapshot();
  });
});
