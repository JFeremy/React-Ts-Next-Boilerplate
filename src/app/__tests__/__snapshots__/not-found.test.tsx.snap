// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component matches snapshot 1`] = `
<div>
  <div>
    <h1>
      ⛔ 404 - Page Not Found ⛔
    </h1>
    <a
      href="/"
    >
      Homepage
    </a>
  </div>
</div>
`;
